cmd/keymasterd/config.go

```diff
@@ -570,6 +547,35 @@ func (state *RuntimeState) setupCertificateManager() error {
 	return nil
 }
 
+func (state *RuntimeState) setupHA() error {
+	if hasDnsLB, err := state.Config.DnsLoadBalancer.HasDNS(); err != nil {
+		return err
+	} else if hasDnsLB {
```

Comment:
better, but I think I mis-spoke... instead of the else if do:
if ! nasDNSLB {
retur nil
}
state.Config.......
....
and you can even cut the last if err!=nil like:
_, err := dnslbcfg.New(state.Config.DnsLoadBalancer, logger)
return err

Reply:
I will still need that last if because the watchdog setup is coming next.

Reply:
ok so here us what I want (so that we have less number of branches... the goal is to leave the function as soon as some condition happens, as that enhances readability


if hasDnsLB, err := state.Config.DnsLoadBalancer.HasDNS(); err != nil {
		return err
	} 
if !hasDnsLB {
   return nil
}

		state.Config.DnsLoadBalancer.DoTLS = true
		if state.Config.DnsLoadBalancer.TcpPort < 1 {
			_, portString, err :=
				net.SplitHostPort(state.Config.Base.AdminAddress)
			if err != nil {
				return err
			}
			port, err := strconv.ParseUint(portString, 10, 16)
			if err != nil {
				return err
			}
			state.Config.DnsLoadBalancer.TcpPort = uint16(port)
			if state.Config.DnsLoadBalancer.FQDN == "" {
				state.Config.DnsLoadBalancer.FQDN =
					state.Config.Base.HostIdentity
			}
		}
		_, err := dnslbcfg.New(state.Config.DnsLoadBalancer, logger)
		if err != nil {
			return err
		}

	return nil
}

Reply:
The change you are suggesting will have to be unwound as soon as the watchdog support goes in, since the watchdog can/should be active regardless of whether DNS LB is enabled.